test/firtool/w0mem.fir

```diff
@@ -0,0 +1,21 @@
+; RUN: firtool %s --format=fir              | circt-opt | FileCheck %s --check-prefix=PARSE
+; RUN: firtool %s --format=fir -disable-opt | circt-opt -pass-pipeline='firrtl.circuit(lower-firrtl-to-rtl)' | FileCheck %s --check-prefix=RTL
```

Comment:
Test case nit-pick: This is changing a fold and the firrtl parser.  Please add unit tests for both pieces to test/firrtl/canonicalization.mlir and test/FIRParser/something.fir.  There's no change to the lower-to-rtl pass, so I don't think you need to test that.

Reply:
Correct solution is to infer the same bus type as firrtl does rather than trying to clean it up after the fact.